snakesist/exist_client.py

```diff
@@ -241,23 +313,24 @@ def retrieve_resources(self, xpath: str) -> List[Resource]:
                       instance supports via its RESTful API)
         :return: The query results as a list of :class:`Resource` objects.
         """
-        try:
-            results_node = self.query(
-                query_expression=f"""for $node in {xpath} return 
-                    <pyexist-result nodeid="{{util:node-id($node)}}" 
-                    absid="{{util:absolute-resource-id($node)}}">
-                    {{$node}}</pyexist-result>"""
-            )
-        except HTTPError:
-            raise ExistAPIError(
-                f"""The attempt to retrieve resources with the expression {xpath}
-                failed. The XPath expression might not be valid."""
-            )
-
-        return [
-            Resource(exist_client=self, query_result=self._parse_item(item))
-            for item in results_node.css_select("pyexist-result")
-        ]
+        xq = (
+            f"for $node in {xpath} "
+            f"return <pyexist-result "
+            f"nodeid='{{util:node-id($node)}}' " 
+            f"absid='{{util:absolute-resource-id($node)}}' "
+            f"path='{{util:collection-name($node) || '/' || util:document-name($node)}}'>"
+            f"{{$node}}</pyexist-result>"
+        )
+        results_node = self.query(xq)
+        resources = []
+        for item in results_node.css_select("pyexist-result"):
+            query_result = self._parse_item(item)
+            if query_result[1] == "1":
```

Comment:
this line's readability would profit from the initially proposed use of a container w/ field names.